File: TangramDotNetLib/TangramForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using TangramCLR;
namespace TangramDotNetLib
{
    public partial class TangramForm : Form
    {
        public string m_strTangramXML = "";
        public TangramForm()
        {
            InitializeComponent();
        }

        private void TangramForm_Load(object sender, EventArgs e)
        {
            Text = "Tangram";
            TangramCLR.TangramCore tangramCore = TangramCLR.TangramCore.GetTangramCore();
            Tangram thisTangram = TangramCore.CreateTangram(this.TangramPanel, this);
            if (thisTangram == null)
            {
                System.Windows.Forms.MessageBox.Show(m_strTangramXML);
            }
            TangramFrame thisFrame = thisTangram.CreateFrame(TangramPanel.Handle, "Default");
            thisFrame.ExtendXml("Default", m_strTangramXML);

        }
    }
}

File: TangramVSIHelper/TangramForm.cpp
#include "TangramForm.h"
using namespace TangramCLR;
namespace TangramVSIHelper {
	void TangramForm::ShowTangram()
	{
		this->Show();
	}

	System::Void TangramForm::TangramForm_Load(System::Object^  sender, System::EventArgs^  e) 
	{
		Text = L"Tangram";
		TangramCLR::TangramCore^ pApp = TangramCore::GetTangramCore();
		TangramCLR::Tangram^ pTangram = pApp->CreateTangram(this->TangramPanel, this);
		if (pTangram != nullptr)
		{
			TangramCLR::TangramFrame^ pFrame = pTangram->CreateFrame(TangramPanel->Handle, L"default");
			if (pFrame != nullptr)
			{
				pFrame->ExtendXml( L"default", m_strTangramXML);
			}
		}
	}
}